internal/revision/scanner.go

```diff
@@ -0,0 +1,124 @@
+package revision
+
+import (
+	"bufio"
+	"io"
+	"unicode"
+)
+
+var zeroRune = rune(0)
+
+// scanner represents a lexical scanner.
+type scanner struct {
```

Comment:
I'm not sure to understand what you mean, only parser use a scanner all thoses functions are internals they are not intended to be used outside this package.